subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/SharedJavaInstallationRegistry.java

```diff
@@ -69,7 +69,7 @@ static SharedJavaInstallationRegistry withLogger(List<InstallationSupplier> supp
     }
 
     private Set<InstallationLocation> collectInstallations(List<InstallationSupplier> suppliers) {
-        return suppliers.stream()
+        return suppliers.parallelStream()
```

Comment:
I wonder if we need to be careful when using the default thread pool as this does. But since this is short lived, maybe that's not an issue

Reply:
I think that's the upside of using parallelStream that you can piggyback on the default ForkJoin pool; managing a separate pool might be enough overhead to not do it in parallel. Using this in combination with the report task, you can see an ~20% improvement by just leveraging a parallel stream (due to us doing IO work in the suppliers)